src/rgw/rgw_sal_motr.h

```diff
@@ -40,6 +44,215 @@ class MotrStore;
 #define RGW_MOTR_BUCKET_HD_IDX_NAME   "motr.rgw.bucket.headers"
 //#define RGW_MOTR_BUCKET_ACL_IDX_NAME  "motr.rgw.bucket.acls"
 
+// Notes for Motr FDMI:
+// (1) Currently FDMI can only trigger events for index operation, not object
+//     operation.
+// (2) When an index op event is generated, the FDMI filter picks up the events
+//     matching the pre-defined rules. The FDMI event contains the key/value of
+//     the index operation. Currently, fdmi can only do sub string matching on
+//     the value part, not key part, and can't specify which index to watch,
+//     the filter rule is applied to all indices.
+//     this 
+// (3) The FDMI event is delivered to an FDMI application which defines a
+//     callback function to process FDMI events.
+//
+// Implementation of object watch-notify using Motr FDMI. 
+// (1) An FDMI filter is added (in Motr configuration yaml file) specifying
+//     a matching sub-string (marker).
+// (2) A notifier creates a set of indices which are used to trigger FDMI
+//     events and to pass messages.
+// (3) For an object, hash(obj_name) is used to pick which index to insert
```

Comment:
Where is this all done, at which process?

Reply:
in rgw_sal_motr.cc::MotrNotifier::notify()
184 int MotrNotifier::notify(const DoutPrefixProvider *dpp, const std::string& key,
185                          MotrWatchNotifyMsg& msg)